java/src/main/native/ai_onnxruntime_OnnxTensor.c

```diff
@@ -136,47 +161,55 @@
 
     // Extract the shape information
-    jboolean mkCopy;
-    jlong* shapeArr = (*jniEnv)->GetLongArrayElements(jniEnv,shape,&mkCopy);
-    jsize shapeLen = (*jniEnv)->GetArrayLength(jniEnv,shape);
+    jlong* shapeArr = (*jniEnv)->GetLongArrayElements(jniEnv, shape, NULL);
+    jsize shapeLen = (*jniEnv)->GetArrayLength(jniEnv, shape);
 
     // Array length
     jsize length = (*jniEnv)->GetArrayLength(jniEnv, stringArr);
 
     // Create the OrtValue
-    OrtValue* ortValue;
-    checkOrtStatus(jniEnv, api, api->CreateTensorAsOrtValue(allocator,(int64_t*)shapeArr,shapeLen,ONNX_TENSOR_ELEMENT_DATA_TYPE_STRING,&ortValue));
-    (*jniEnv)->ReleaseLongArrayElements(jniEnv,shape,shapeArr,JNI_ABORT);
-
-    // Create the buffers for the Java strings
-    const char** strings;
-    checkOrtStatus(jniEnv, api, api->AllocatorAlloc(allocator,sizeof(char*)*length,(void**)&strings));
-    jobject* javaStrings;
-    checkOrtStatus(jniEnv, api, api->AllocatorAlloc(allocator,sizeof(jobject)*length,(void**)&javaStrings));
-
-    // Copy the java strings into the buffers
-    for (jsize i = 0; i < length; i++) {
-        javaStrings[i] = (*jniEnv)->GetObjectArrayElement(jniEnv,stringArr,i);
-        strings[i] = (*jniEnv)->GetStringUTFChars(jniEnv,javaStrings[i],NULL);
-    }
-
-    // Assign the strings into the Tensor
-    checkOrtStatus(jniEnv, api, api->FillStringTensor(ortValue,strings,length));
-
-    // Release the Java strings
-    for (int i = 0; i < length; i++) {
-        (*jniEnv)->ReleaseStringUTFChars(jniEnv,javaStrings[i],strings[i]);
+    OrtValue* ortValue = NULL;
+    OrtErrorCode code = checkOrtStatus(jniEnv, api, api->CreateTensorAsOrtValue(allocator, (int64_t*)shapeArr, shapeLen,
+                                        ONNX_TENSOR_ELEMENT_DATA_TYPE_STRING, &ortValue));
+    (*jniEnv)->ReleaseLongArrayElements(jniEnv, shape, shapeArr, JNI_ABORT);
+
+    if (code == ORT_OK) {
+      // Create the buffers for the Java strings
+      const char** strings = NULL;
+      code = checkOrtStatus(jniEnv, api, api->AllocatorAlloc(allocator, sizeof(char*) * length, (void**)&strings));
+
+      if (code == ORT_OK) {
+        // Copy the java strings into the buffers
+        for (jsize i = 0; i < length; i++) {
+            jobject javaString = (*jniEnv)->GetObjectArrayElement(jniEnv, stringArr, i);
+            strings[i] = (*jniEnv)->GetStringUTFChars(jniEnv, javaString, NULL);
+        }
+
+        // Assign the strings into the Tensor
+        code = checkOrtStatus(jniEnv, api, api->FillStringTensor(ortValue, strings, length));
+
+        // Release the Java strings
+        for (int i = 0; i < length; i++) {
+            jobject javaString = (*jniEnv)->GetObjectArrayElement(jniEnv, stringArr, i);
+            (*jniEnv)->ReleaseStringUTFChars(jniEnv, javaString, strings[i]);
+        }
+
+        // Release the buffers
+        checkOrtStatus(jniEnv, api, api->AllocatorFree(allocator, (void*)strings));
```

Comment:
>checkOrtStatus [](http://example.com/codeflow?start=8&length=14)

Here, we may record an exception, but the OrtValue is not freed as a result since the code is not checked.

Reply:
Ok, I'll add a check. Here we might swallow any exception that comes out of `AllocatorFree` if the `FillStringTensor` call failed, but we still want to free that memory if that call failed.

Reply:
I am in favor or omitting an error check on free.

Reply:
Ok